src/Common/ProxyConfigurationResolverProvider.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <base/types.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Common/ProxyConfigurationResolver.h>
+
+namespace DB
+{
+
+/*
+ * Returns appropriate ProxyConfigurationResolver based on current CH settings (Remote resolver or List resolver).
+ * If no configuration is found, returns Environment Resolver.
+ * */
+class ProxyConfigurationResolverProvider
+{
+public:
+    static std::shared_ptr<ProxyConfigurationResolver> get();
```

Comment:
```suggestion
    // Get default proxy configuration from global `Context`'s config or from ENV if the global `Config` is missing
    static std::shared_ptr<ProxyConfigurationResolver> get();
```

Reply:
Do you think it's actually necessary? I have already put a general comment on top of the class.

```
/*
 * Returns appropriate ProxyConfigurationResolver based on current CH settings (Remote resolver or List resolver).
 * If no configuration is found, returns Environment Resolver.
 * */
```

Nit: Your suggestion says "get default proxy configuration", but what it actually does is return a resolver and not the configuration itself.